Initial Galaxy A52 and A72 support

frostg-012 opened the pull request:
a52q will be maintained by TestaMic
a72q will be maintained by frostg012

Known issue:
- Disabled HDR for front camera recording, untill we found fix for that.

Co-authored-by: TestaMic <mateidiac9@gmail.com>

arter97 committed:
sm7125: hfr: Relax timers for automatic refresh rate switching

Due to visible gamma shifts on our panels, we should make the switchings
to happen conservatively.

With idle timer >= touch timer, idle timer now always takes precedence,
but we're leaving touch timer property to immediately switch back to
90 Hz on touch input events.

Display power timer is left untouched as it's responsible for display
mode changes such as Doze.

Change-Id: If8b1e00a2d2213c4d2288ab28fad4ba0826a36e8
Signed-off-by: Park Ju Hyung <qkrwngud825@gmail.com>
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

yagzie committed:
a52q/a72q: camera: disable hdr/vdis on video

Breaks video recording, keep this until another solution is found.

Change-Id: Ic5d3f9dd57c90b8e8044a413a6f44ca5f98380e0

frostg-012 committed:
a52q/a72q: patches: Add custom kernel on build

- based on PROCA patch script.

Change-Id: I4cf5b8108f1aa27638e61053f79a11de31cb5adc
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
a52q/a72q: Add device compatibility_matrix from a52sxq

[1] - https://github.com/salvogiangri/UN1CA/commit/ed2c8ff5919d40d3ba201b6ce9dd3c60137d7296
Change-Id: I2555ff95ca346b9eb556509029b6a88eb4000424
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
sm7125: display: Sync display props with QCOM common

Change-Id: Ib0c75e4eb4c09b35c2a8585404a13e2e12f9f73f
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
sm7125: miscs: Uprev radio HAL to 1.6

Change-Id: I40438ad781f9becdfa13a0745bbdb3d838ff2c90
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

pascua28 committed:
sm7125: patches: Switch to AIDL display HAL

Change-Id: Ibf810dbd4856aa734be1a56f90b442187936fff0
Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
a52q/a72q: patches: Push QDCM calibration from stock

Change-Id: Ic10b1399ca43f2885111fc70e700cd60efe7773c
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

pascua28 committed:
a52q/a72q: display: Append RenderIntent to QDCM calibration

Change-Id: I965b36c5ea4735c49231046822c3d4034f04be4c
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
sm7125: camera: Replace MIDAS and singletake configs with a73xqxx

Change-Id: I62b85077a99c447edae5efe31377ad247ec97d9d
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
a52q/a72q: camera: Fix MIDAS model detection

Change-Id: I39a96dadfe0f592f85ae2c9f527b267facf7b8e7
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

salvogiangri committed:
sm7125: hfr: init

Change-Id: I00f45980b2da9928a095ca020b8f2e4451dd790e
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

salvogiangri committed:
sm7125: patches: add required camera libs

Change-Id: I70c6be0ce1bd41062a9270b9cf8cab693d8511bd
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
platforms: sm7125: init

Co-authored-by: Salvo Giangreco <giangrecosalvo9@gmail.com>
Change-Id: Ibc77569c3611fdba855965b64864190116501a25
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
target: a72q: init

Change-Id: I451530c7d985da98197b35ef2d6058cdd0ce3f0b

frostg-012 committed:
target: a52q: init

Co-authored-by: TestaMic <mateidiac9@gmail.com>
Change-Id: I0542b3602eae89a136e950ce17fab8b57cef8484
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

pascua28 committed:
sm7125: display: Apply hex patch for libsdmutils.so

Change-Id: I4f1cec93e7c5474aeafb8a95b8daef804e648110
Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

salvogiangri committed:
a52q/a72q: patches: Fix Edge lighting corner radius

Change-Id: I1cd1b022966041e02f745f9907f1cf0fe2afcc96
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

salvogiangri committed:
a52q/a72q: installer: init

Change-Id: Ied39b31de168ddb41b2a7e5acdd10ae1a3149207
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

frostg-012 committed:
sm7125: display: vendor prop tweaks

Change-Id: I3f0b90b0a8fb2a79b98febc3a818798927e0ae02
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

pascua28 committed:
prebuilts/samsung: a73xqxx: add display stack

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>

fishiatee commented:
Any chance for this to be extended to A52 5G (a52xq) devices too?

the-mck commented:
Good job guys

matei9 commented:
> Any chance for this to be extended to A52 5G (a52xq) devices too?

a52sxq already has official UN1CA support, check github releases of this repo.

frostg-012 commented:
> > Any chance for this to be extended to A52 5G (a52xq) devices too?
> 
> a52sxq already has official UN1CA support, check github releases of this repo.

he want a52xq, not a52sxq

matei9 commented:
> > > Any chance for this to be extended to A52 5G (a52xq) devices too?
> > 
> > 
> > a52sxq already has official UN1CA support, check github releases of this repo.
> 
> he want a52xq, not a52sxq

Oh right, sorry.
was in a hurry when i read it.

kishan-dhankecha commented:
Any timeline for this? I mean when we will be able to get this and install it

matei9 commented:
> Any timeline for this? I mean when we will be able to get this and install it

If you don't mind installing a beta, you can go in the official telegram of UN1CA, and you will find beta builds for a52q/a72q.

kishan-dhankecha commented:
I can intsall beta and test as well. Just a small question will I have to wipe my data? I have ROM version 2.5.6 Eureka installed right now

pascua28 committed:
sm7125: display: fix `GetCustomDimensions()` error

Fix `GetCustomDimensions()` error logspam from graphics mapper caused by
early exit after calling `getMetaData()` especially when using newer
adreno drivers.

No functional change.

Signed-off-by: Samuel Pascua <pascua.samuel.14@gmail.com>
Signed-off-by: Vayruz Rafli <mikhailovnakowalski@gmail.com>

matei9 commented:
> I can intsall beta and test as well. Just a small question will I have to wipe my data? I have ROM version 2.5.6 Eureka installed right now

Dirty flash might work, but clean flashing will give you the best experience.

majaahh commented on target/a52q/patches/kernel/customize.sh:
```diff
@@ -0,0 +1,39 @@
+
+MKBOOTIMG_ARGS="$(unpack_bootimg --boot_img "$TMP_DIR/boot.img" --out "$TMP_DIR/out" --format mkbootimg 2>&1)"
+
+[ ! -f "$TMP_DIR/out/kernel" ] && ABORT "Failed to extract boot.img\n\n$MKBOOTIMG_ARGS"
```
Like in https://github.com/salvogiangri/UN1CA/pull/562/changes/2633367c9b204fe8ad0e1ecf2eb0b6873c90caf0#r2740842955 I suggest
```
if [[ ! -f "$TMP_DIR/out/kernel" ]]; then
    LOGE "Failed to extract boot.img\n\n$MKBOOTIMG_ARGS"
    return 1
fi
```
